src/gamestate/mod.rs

```diff
@@ -450,7 +450,13 @@ impl GameState {
                         )?;
                 }
                 "soldieradvice" => {
-                    // I think they removed this
+                    other_player
+                        .get_or_insert_with(Default::default)
+                        .soldier_advice =
+                        match val.into::<i16>("other player soldier advice")? {
+                            ..=-1 => None,
+                            x => Some(x.try_into().unwrap_or(1)),
```

Comment:
Can this just be `val.into::<u16>("other player soldier advice").ok()`, or is a negative value expected here and should therefore not be logged as an error?

Reply:
i dont know for sure whether it can be negative but when i was checking it, sometimes weird numbers came out in the thousands thats why i precautiously check for negative numbers. but we can also change it  since i have not seen a negative number yet

Reply:
In the thousands? That makes me think they do weird bitshifting again and this is actually two values encoded in one int. If that is the case this should probably be masked `(x & 0xFF)`. Do you still have these large values somewhere? I would love to check what they might be encoding in there. Otherwise I will just try to repo this myself tomorrow

Reply:
Unfortunately i don't but either way we can simplify it changing it later is no big deal in this case i ran all my characters through the change seems fine so far

Reply:
@the-marenga 

Apologies for sending you the entire otherPlayer data at once, but I noticed this in my debug log and remembered that you wanted to review it. In this particular case, the SoldierAdvice value being 1000 makes sense because I'm using level 1 soldiers. However, the last instance I observed had values of 16000 and 1, which seemed unusual.

otherplayergroupname.r:Bronxstreet&otherplayer.playerlookat:24742/0/50/111099/301715/5213/47983/0/3/401/407/3/403/1/3/4/1/0/4/1/9/50/79/248/250/127/87/87/559/424/426/0/0/0/0/0/0/0/0/6/2052/97/0/23/0/0/36/0/0/6304/0/3/2063/135/0/6/0/0/39/0/0/8201/0/1572869/2050/260/0/23/0/0/51/0/0/9857/2228224/2097156/2010/155/0/3/0/0/74/0/0/8408/2555904/65544/53/0/0/6/0/0/48/0/0/74520/0/1507335/2050/152/0/23/0/0/56/0/0/47532/2162688/9/5/0/0/3/4/0/33/25/0/1763/0/2162698/20/0/0/3/0/0/85/0/0/5362/2490368/2228225/1052/125/271/23/0/0/98/0/0/109320/2097152/0/0/0/0/0/0/0/0/0/0/0/0/0/0/1927/-1/10393/0/0/1668861919/0/799/125/271/25500/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/0/8/7/8/8/7/8/8/7/7/7/8/7/0/0/0/0/0/0/0/0/9266/2830/21/0/0/1673958341/1676088540/1676088540/0/0/0/12750/4250/12750/4250/0/0/0/0/162/1003/25/25/25/0/0/0/0/0/0/1/0/0&otherdescription.s:&otherplayername.r:Daveone&otherplayerunitlevel(4):50/25/25/25/&otherplayerfriendstatus:0&otherplayerfortressrank:40375&otherplayerpetbonus.petbonus:24742/0/0/0/0/0&soldieradvice:1000